From 00abf5d42c9b9c979b7ac69758a8f6745744f929 Mon Sep 17 00:00:00 2001
From: Badlop <badlop@process-one.net>
Date: Wed, 26 Feb 2020 13:35:47 +0100
Subject: [PATCH] Fix handle of 'http' atom in Headers, problem introduced in
 357e7e11

---
 src/ejabberd_http.erl | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/ejabberd_http.erl b/src/ejabberd_http.erl
index fdc9321ef3..fad51f44fc 100644
--- a/src/ejabberd_http.erl
+++ b/src/ejabberd_http.erl
@@ -857,9 +857,13 @@ parse_urlencoded(<<>>, Last, Cur, _State) ->
 parse_urlencoded(undefined, _, _, _) -> [].
 
 apply_custom_headers(Headers, CustomHeaders) ->
-    M = maps:merge(maps:from_list(Headers),
+    {Doctype, Headers2} = case Headers -- [html] of
+	Headers -> {[], Headers};
+	Other -> {[html], Other}
+    end,
+    M = maps:merge(maps:from_list(Headers2),
 		   maps:from_list(CustomHeaders)),
-    maps:to_list(M).
+    Doctype ++ maps:to_list(M).
 
 % The following code is mostly taken from yaws_ssl.erl
 
