diff -Nru EasyRSA-3.0.5.orig/easyrsa EasyRSA-3.0.5/easyrsa
--- EasyRSA-3.0.5.orig/easyrsa	2018-09-15 07:21:19.000000000 +0300
+++ EasyRSA-3.0.5/easyrsa	2018-12-19 02:02:33.194000000 +0300
@@ -131,8 +131,9 @@
   export-p12 <filename_base> [ cmd-opts ]
       Export a PKCS#12 file with the keypair specified by <filename_base>"
 			opts="
-        noca  - do not include the ca.crt file in the PKCS12 output
-        nokey - do not include the private key in the PKCS12 output" ;;
+        noca   - do not include the ca.crt file in the PKCS12 output
+        nokey  - do not include the private key in the PKCS12 output
+        nopass - use blank password" ;;
 		export-p7) text="
   export-p7 <filename_base> [ cmd-opts ]
       Export a PKCS#7 file with the pubkey specified by <filename_base>"
@@ -919,10 +920,12 @@
 	# opts support
 	want_ca=1
 	want_key=1
+	want_pass=1
 	while [ -n "$1" ]; do
 		case "$1" in
 			noca) want_ca= ;;
 			nokey) want_key= ;;
+			nopass) want_pass= ;;
 			*) warn "Ignoring unknown command option: '$1'" ;;
 		esac
 		shift
@@ -954,10 +957,14 @@
 			pkcs_opts="$pkcs_opts -nokeys"
 		fi
 
+		if [ ! $want_pass ]; then
+			pkcs_opts="$pkcs_opts -password pass:"
+		fi
+
 		# export the p12:
 		# shellcheck disable=SC2086
 		"$EASYRSA_OPENSSL" pkcs12 -in "$crt_in" -inkey "$key_in" -export \
-			-out "$pkcs_out" $pkcs_opts || die "\
+			-name "$short_name" -out "$pkcs_out" $pkcs_opts || die "\
 Export of p12 failed: see above for related openssl errors."
 	;;
 	p7)
