diff -Naur a/sapi/fpm/fpm/fpm_main.c b/sapi/fpm/fpm/fpm_main.c
--- a/sapi/fpm/fpm/fpm_main.c	2018-07-31 16:33:06.000000000 +0200
+++ b/sapi/fpm/fpm/fpm_main.c	2018-08-03 19:33:15.108465895 +0200
@@ -1335,13 +1335,14 @@
 	/* INI stuff */
 	ini = FCGI_GETENV(request, "PHP_VALUE");
 	if (ini) {
-		int mode = ZEND_INI_USER;
+		int mode = ZEND_INI_PERDIR;
 		char *tmp;
 		spprintf(&tmp, 0, "%s\n", ini);
 		zend_parse_ini_string(tmp, 1, ZEND_INI_SCANNER_NORMAL, (zend_ini_parser_cb_t)fastcgi_ini_parser, &mode);
 		efree(tmp);
 	}
 
+#if 0
 	ini = FCGI_GETENV(request, "PHP_ADMIN_VALUE");
 	if (ini) {
 		int mode = ZEND_INI_SYSTEM;
@@ -1350,6 +1351,7 @@
 		zend_parse_ini_string(tmp, 1, ZEND_INI_SCANNER_NORMAL, (zend_ini_parser_cb_t)fastcgi_ini_parser, &mode);
 		efree(tmp);
 	}
+#endif
 }
 /* }}} */
 
