diff -Naur php-7.2.0.orig/sapi/fpm/fpm/fpm_main.c php-7.2.0/sapi/fpm/fpm/fpm_main.c
--- php-7.2.0.orig/sapi/fpm/fpm/fpm_main.c	2017-11-12 18:58:10.347321756 +0100
+++ php-7.2.0/sapi/fpm/fpm/fpm_main.c	2017-11-12 18:58:23.170173604 +0100
@@ -1400,13 +1400,14 @@
 	/* INI stuff */
 	ini = FCGI_GETENV(request, "PHP_VALUE");
 	if (ini) {
-		int mode = ZEND_INI_USER;
+		int mode = ZEND_INI_PERDIR;
 		char *tmp;
 		spprintf(&tmp, 0, "%s\n", ini);
 		zend_parse_ini_string(tmp, 1, ZEND_INI_SCANNER_NORMAL, (zend_ini_parser_cb_t)fastcgi_ini_parser, &mode);
 		efree(tmp);
 	}
 
+#if 0
 	ini = FCGI_GETENV(request, "PHP_ADMIN_VALUE");
 	if (ini) {
 		int mode = ZEND_INI_SYSTEM;
@@ -1415,6 +1416,7 @@
 		zend_parse_ini_string(tmp, 1, ZEND_INI_SCANNER_NORMAL, (zend_ini_parser_cb_t)fastcgi_ini_parser, &mode);
 		efree(tmp);
 	}
+#endif
 }
 /* }}} */
 
