diff -urpN opensmtpd-7.3.0p1.bak/openbsd-compat/libasr/asr.c opensmtpd-7.3.0p1/openbsd-compat/libasr/asr.c
--- opensmtpd-7.3.0p1.bak/openbsd-compat/libasr/asr.c	2023-06-27 17:04:38.000000000 +0300
+++ opensmtpd-7.3.0p1/openbsd-compat/libasr/asr.c	2024-02-06 11:51:32.800833068 +0300
@@ -537,8 +537,13 @@ asr_ctx_create(void)
 	ac->ac_options = RES_RECURSE | RES_DEFNAMES | RES_DNSRCH;
 	ac->ac_refcount = 1;
 	ac->ac_ndots = 1;
+#ifndef ASR_IPV4_BEFORE_IPV6
+	ac->ac_family[0] = AF_INET6;
+	ac->ac_family[1] = AF_INET;
+#else
 	ac->ac_family[0] = AF_INET;
 	ac->ac_family[1] = AF_INET6;
+#endif
 	ac->ac_family[2] = -1;
 
 	ac->ac_nscount = 0;
